### PR TITLE
Fix type inference for GPU scalars

### PR DESCRIPTION
**Pull request by orausch**

### CODE REVIEW

**Comment by orausch** [comment]
This fix seems to work, but I'm not very confident that it covers all cases, especially since there are still many similar statements in the CPU codegen:
```
$ git grep 'isinstance.*pointer'
dace/codegen/targets/cpp.py:    is_scalar = not isinstance(conntype, dtypes.pointer) and desc.storage is not dtypes.StorageType.GPU_Global
dace/codegen/targets/cpu.py:        if isinstance(dtype, dtypes.pointer):
dace/codegen/targets/cpu.py:                is_scalar = not isinstance(conntype, dtypes.pointer)
dace/codegen/targets/cpu.py:        is_scalar = not isinstance(conntype, dtypes.pointer)
dace/codegen/targets/cpu.py:        is_pointer = isinstance(conntype, dtypes.pointer)
dace/codegen/targets/cpu.py:            if isinstance(cdtype, dtypes.pointer):
dace/codegen/targets/cpu.py:        elif isinstance(cdtype, dtypes.pointer):
dace/codegen/targets/cuda.py:                        if isinstance(field_type, dtypes.pointer):
```

File: tests/library/addlib/__init__.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
```
Comment (tbennun) [inline tests/library/addlib/__init__.py:1]:
```suggestion
# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
```

File: tests/library/scalars_gpu_test.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import pytest
```
Comment (tbennun) [inline tests/library/scalars_gpu_test.py:1]:
```suggestion
# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
import pytest
```

**Review by tbennun — CHANGES_REQUESTED** [review]

File: tests/library/scalars_gpu_test.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        infer_types.infer_connector_types(sdfg)
+    else:
+        infer_types.infer_connector_types(sdfg)
+        sdfg.expand_library_nodes()
+        infer_types.infer_connector_types(sdfg)
+
+    input_arr = np.array([1]).astype(np.float32) if input_array else 1
+    output_arr = np.array([0]).astype(np.float32)
+    sdfg(input_arr=input_arr, output_arr=output_arr)
+    assert output_arr[0] == 2
```
Comment (tbennun) [inline tests/library/scalars_gpu_test.py:61]: Missing `__main__`

**Review by tbennun — APPROVED** [review]

**Pull request merged**